tests/disk.rs

```diff
@@ -0,0 +1,62 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+#[test]
+#[cfg(all(feature = "system", feature = "disk"))]
+fn test_disks() {
+    if sysinfo::IS_SUPPORTED_SYSTEM {
+        let s = sysinfo::System::new_all();
+        // If we don't have any physical core present, it's very likely that we're inside a VM...
+        if s.physical_core_count().unwrap_or_default() > 0 {
+            let mut disks = sysinfo::Disks::new();
+            assert!(disks.list().is_empty());
+            disks.refresh_list();
+            assert!(!disks.list().is_empty());
+        }
+    }
+}
+
+#[test]
+#[cfg(feature = "disk")]
+fn test_disks_usage() {
+    use std::io::Write;
+    use tempfile::NamedTempFile;
+
+    if !sysinfo::IS_SUPPORTED_SYSTEM {
+        return;
+    }
+
+    let mut disks = sysinfo::Disks::new_with_refreshed_list();
+
+    let mut file = NamedTempFile::new().unwrap();
+
+    // Write 10mb worth of data to the temp file. Repeat this 10 times to increase the chances
+    // of the OS registering the disk writes.
+    for _ in 0..10 {
+        let data = vec![1u8; 10 * 1024 * 1024];
```

Comment:
Can be moved outside of the loop I think, no need to reallocate 10MB of memory at every turn.